### DISCUSSION TITLE
Include Schema in schema create an Array, why ?

### DISCUSSION THREAD

**Discussion by davy-blavette** [Automattic/mongoose#11818] [General]
Hi,

Why I have a difference with :

```
dataExpression: {
      colere: [expressionSchema],
      degout: [expressionSchema],
      peur: [expressionSchema],
      joie: [expressionSchema],
      triste: [expressionSchema],
      surprise: [expressionSchema],
      neutre: [expressionSchema],
    },
```

result is dataExpression: { 
colere: []
} is good, I have an Object....

If I use :
dataExpression: [dataExpressionSchema]

with :

```
const dataExpressionSchema = new Schema({
    colere: [expressionSchema],
    degout: [expressionSchema],
    peur: [expressionSchema],
    joie: [expressionSchema],
    triste: [expressionSchema],
    surprise: [expressionSchema],
    neutre: [expressionSchema],
});
```

result is dataExpression: **[**
{ colere: []}
**]** is bad because I have an Array....

**Comment by jmtt89** [comment]
because your using find method to your queries, find always return a array, if you want only one element need to use findOne or FindById

**Reply by davy-blavette** [reply]
This is a schema that I use as a model, and my query is :
let result = await Trainer.findByIdAndUpdate(id, req.body);
req.body is object{object} and I have a return object[{object}] if I use dataExpression: [dataExpressionSchema]...I feel like wrapping new schemas creates an array. I found the solution without using a new schema :

    dataExpression: dataExpressionSchema,

and...

const dataExpressionSchema = {
    colere: [expressionSchema],
    degout: [expressionSchema],
    peur: [expressionSchema],
    joie: [expressionSchema],
    triste: [expressionSchema],
    surprise: [expressionSchema],
    neutre: [expressionSchema],
};

👍
